File: app/src/main/java/zebrostudio/wallr100/WallRApplication.java
package zebrostudio.wallr100;

import android.content.Context;
import android.support.multidex.MultiDex;

import javax.inject.Inject;
import javax.inject.Singleton;

import dagger.android.AndroidInjector;
import dagger.android.support.DaggerApplication;
import zebrostudio.wallr100.data.DataManager;
import zebrostudio.wallr100.di.component.AppComponent;
import zebrostudio.wallr100.di.component.DaggerAppComponent;

@Singleton
public class WallRApplication extends DaggerApplication {

    @Inject
    DataManager mDataManager;

    @Override
    public void onCreate() {
        super.onCreate();
        mDataManager.requestFirebasePersistenceInitialization(this);
        mDataManager.requestOneSignalSdkInitialization(this);
        mDataManager.requestGlideTagConfiguration();
        mDataManager.requestAutomaticWallpaperChangerJobInitialization();
    }

    @Override
    protected AndroidInjector<? extends DaggerApplication> applicationInjector() {
        AppComponent appComponent = DaggerAppComponent.builder().application(this).build();
        appComponent.inject(this);
        return appComponent;
    }

    @Override
    protected void attachBaseContext(Context base) {
        super.attachBaseContext(base);
        MultiDex.install(this);
    }

}

File: app/src/main/java/zebrostudio/wallr100/WallrApplication.kt
package zebrostudio.wallr100

import android.app.Activity
import android.app.Application
import com.bumptech.glide.request.target.ViewTarget
import dagger.android.DispatchingAndroidInjector
import dagger.android.HasActivityInjector
import zebrostudio.wallr100.di.DaggerAppComponent
import javax.inject.Inject

class WallrApplication : Application(), HasActivityInjector {

  @Inject
  internal lateinit var activityDispatchingAndroidInjector: DispatchingAndroidInjector<Activity>

  override fun onCreate() {
    super.onCreate()
    DaggerAppComponent.builder()
        .application(this)
        .build()
        .inject(this)
    ViewTarget.setTagId(R.id.glide_tag)
  }

  override fun activityInjector() = activityDispatchingAndroidInjector

}